### ISSUE TITLE
Date and time calculations and formating of results.

### ISSUE THREAD

**Issue by fr4nk3nst3in** [metabase/metabase#12016]
Please add a possibility to make any date and time calculations.

Suppose we have two timestamp or datetime fields, and we want to get the difference between this tow fields. One of use case is for example reaction time of a newsletter that was sent at one specific datetime observation and I want see the cooling of this newsletter. That means how many views/clicks was made since 1,2,3...n day/hours.

An SQL Solution is a possible solution, but a native solution would be nice.

And give a possibility to format this calculation in a reasonable manner.

e.g. result of an SQL statement with a proper format of duration
<img width="406" alt="Снимок экрана 2020-03-01 в 12 23 33" src="https://user-images.githubusercontent.com/4451184/75624762-89f68a00-5bb7-11ea-8dff-64547aeddffe.png">
And this is what I get as a result.
<img width="1416" alt="Снимок экрана 2020-03-01 в 12 25 24" src="https://user-images.githubusercontent.com/4451184/75624786-d93cba80-5bb7-11ea-9968-ec52fcb8946d.png">
There is no possibility to format the duration as $HOUR:$MINUTE for example.

**Comment by flamber** [comment]
Hi @fr4nk3nst3in 
Looks like a duplicate of #11330 - upvote by clicking :+1: on the first post.
As for the formatting, I'm not sure what's going on - and there's a spell mistake too - is that from Metabase or did you do something? But that should have it's own issue with steps to reproduce.

**Comment by fr4nk3nst3in** [comment]
The search does not give me result with named issue. So it is indeed a duplicate.

That I mean, is by using and SQL IDE I get a nice result with a format of %H:%M (hour, minute). Metabase does detect correctly, that the result is a timestamp. Unfortunately there is no way to format this result as %H:%M like normal datetime formatting in metabase.

**Comment by flamber** [comment]
@fr4nk3nst3in Sounds like #11427 - but I still cannot reproduce what you were seeing and the spell mistake too. Can you provide steps to reproduce?

**Comment by fr4nk3nst3in** [comment]
@flamber Yes. It is that I mean. But if I am using custom SQL for aggregation, due to missing possibility to summarize timestamps, and usage a number as visualization type, metabase detects the result as a timestamp and shows complete string with years, months etc. If I click on visualization type usually, there are options for formatting a value. In my case there are no options there. The options should be like in #11427 with the possibility to say do not show the date part of a timestamp and show only hours and minutes.

**Comment by flamber** [comment]
@fr4nk3nst3in Okay, I cannot reproduce it. Without steps to reproduce it's very difficult to figure out if it's an existing issue or a new. Or if there's a workaround for it.

**Comment by fr4nk3nst3in** [comment]
@flamber the workaround is to cast the value as character.

**Comment by fr4nk3nst3in** [comment]
@flamber here is sample data http://sqlfiddle.com/#!15/da7593/2

Suppose we have a schema with a table
```
create table t (
id integer,
duration interval
);

insert into t (id, duration)
values (1,'00:10:00'),(2,'00:20:00'),(3,'00:30:00');
```

Then let us visualize following SQL Query:

```
select sum(duration)
from t;
```

This is the result:
<img width="1460" alt="Снимок экрана 2020-03-01 в 15 59 16" src="https://user-images.githubusercontent.com/4451184/75628080-f2ecfa80-5bd5-11ea-9e67-9ab0bcd15418.png">

A complete string with year, months etc. After clicking of a option for the number,  there are no options.

<img width="722" alt="Снимок экрана 2020-03-01 в 15 59 31" src="https://user-images.githubusercontent.com/4451184/75628110-1e6fe500-5bd6-11ea-8302-80b69a59a8e1.png">

And now with a cast to a string.

```
select to_char(sum(duration), 'HH24:MM')
from t;

```

gives us desired format for the interval.
<img width="1180" alt="Снимок экрана 2020-03-01 в 16 00 11" src="https://user-images.githubusercontent.com/4451184/75628122-3f383a80-5bd6-11ea-9ee1-78262b55e425.png">

The issue is that where are no options for formatting in the manner as described in #11427.

**Comment by flamber** [comment]
@fr4nk3nst3in Ahhh, now I see. It's a combination of multiple issues. Because `interval` isn't registered as a datetime type, but just as a string, then formatting options isn't available #2656
